### ISSUE TITLE
Not compiling for Arnold 7.2.4.1

### ISSUE THREAD

**Issue by lozalizx** [mruegenberg/mrAiHouOceanShader#3]
Hi there!

We have successfully built this with Arnold-6.2.1.1, after an Arnold version update on our system due to bugs in that version we are trying to build this with Arnold-7.2.4.1 but getting the following errors when trying to run the compile.sh script

```src/ai_ocean_samplelayers.cpp:44:5: error: ‘AtCritSec’ does not name a type
     AtCritSec critSec;
     ^~~~~~~~~
src/ai_ocean_samplelayers.cpp: In function ‘void Initialize(AtRenderSession*, AtNode*)’:
src/ai_ocean_samplelayers.cpp:72:27: error: ‘struct ShaderData’ has no member named ‘critSec’
     AiCritSecInit(&(data->critSec));
                           ^~~~~~~
src/ai_ocean_samplelayers.cpp:72:35: error: ‘AiCritSecInit’ was not declared in this scope
     AiCritSecInit(&(data->critSec));
                                   ^
src/ai_ocean_samplelayers.cpp: In function ‘void Finish(AtNode*)’:
src/ai_ocean_samplelayers.cpp:95:32: error: ‘struct ShaderData’ has no member named ‘critSec’
         AiCritSecClose(&(data->critSec));
                                ^~~~~~~
src/ai_ocean_samplelayers.cpp:95:40: error: ‘AiCritSecClose’ was not declared in this scope
         AiCritSecClose(&(data->critSec));
                                        ^
src/ai_ocean_samplelayers.cpp: In function ‘void Evaluate(AtNode*, AtShaderGlobals*)’:
src/ai_ocean_samplelayers.cpp:131:32: error: ‘struct ShaderData’ has no member named ‘critSec’
         AiCritSecEnter(&(data->critSec)); // tmpnam is not threadsafe
                                ^~~~~~~
src/ai_ocean_samplelayers.cpp:131:40: error: ‘AiCritSecEnter’ was not declared in this scope
         AiCritSecEnter(&(data->critSec)); // tmpnam is not threadsafe
                                        ^
src/ai_ocean_samplelayers.cpp:160:32: error: ‘struct ShaderData’ has no member named ‘critSec’
         AiCritSecLeave(&(data->critSec));
                                ^~~~~~~
src/ai_ocean_samplelayers.cpp:160:40: error: ‘AiCritSecLeave’ was not declared in this scope
         AiCritSecLeave(&(data->critSec));
                                        ^
```
I have put the later arnold sdk in the /deps folder and updated the houdini version to 18.5.759. 

Maybe this just isn't compatible with Arnold 7 but thought i'd reach out and check.

Thanks!

-L

**Comment by lozalizx** [comment]
After a bit more digging looks like Arnold deprecated all of the following in arnold 7:

```  // replace with AtMutex
 void AiCritSecInit(AtCritSec* cs);
 void AiCritSecInitRecursive(AtCritSec* cs);
 void AiCritSecClose(AtCritSec* cs);
 void AiCritSecEnter(AtCritSec* cs);
 void AiCritSecLeave(AtCritSec* cs);
``` 
details here: https://help.autodesk.com/view/ARNOL/ENU/?guid=arnold_core_7200_html so the code likely needs updating to use AtMutex as they suggest to work with Arnold 7

**Comment by mruegenberg** [comment]
Hey, thanks for the bugreport and fix! Closing this as #4 is merged
